README.md

```diff
@@ -1 +1,54 @@
+but tried to make it clear, tested, logged and  without duplications. But it still can't be considered as some best
+practices.
+
+## Overview
+
+### Business process description
+Project consists of several services communicating to ensure proper execution of some business processes. TODO sukhoa : 
+here should be description of the system.
+
+### Architecture overview
+Typically each individual service is just a small monolith or hexagonal application responsible for data and operations 
+from some certain business area. For example order management, user management etc. Service may have a number of inbound
+channels and outbound channels. Inbound channels enables service to get some data from the outside world. It may be 
+different types of data - client request, some event, resource state or whatever. Inbound channels in our system are 
+represented by REST API of the service and the queues or topics on which the service is subscribed. The REST API calls 
+may sometimes overlay or simple duplicate the messaging API logic. In our particular case it have been made for testing 
+purposes and making developer’s life easier. We use Swagger to consume our REST API during developing. So REST API and 
+messaging API should be designed to be consistent - REST calls and equivalent messaging call should always lead to same 
+results in system and produce the same response. We achieve this by always delegating all the business operations to 
+the services layer. Data from inbound channels regardless of Its nature are processed by the services (we mean internal 
+services, DDD term) of the microservice. Only services are responsible for all the business logic and there is no way 
+to work around them. The only thing the REST API and messaging API layer is responsible is to transform data into the 
+form which may be fed to services. Note that equivalent API not necessarily have the same incoming data - just because 
+the messaging API receives the messages and the REST API may receive some dto which may have slightly different structure
+and some path/request parameters.
+
+Outbound channels - queues and topics to where we publish our outcome events and messages and our local datastore.
+
+![Typical microservice architecture](course-materials/git-images/typical-service-arch.png)
+
+Services _don’t_ have any shared entities, data structures and each service has its own schema in our Postgres instance 
+which enables us:
+
+1. Deploy services and migrate service’s schema and data independently
+1. Switch to another DBMS (Polyglot persistence approach) at any time we decide without any impact to other services.
+
+Services typically consist of two modules:
+1. **app** module contains the implementation of service - properties,  services, controllers, dao objects, domain and dto 
+classes, utility classes. It builds into some jar or war file which might be executed or deployed to some web server.
```

Comment:
It is assembled into JAR/WAR file which might be executed or deployed to web/application server.